backend/internal/database/book.go

```diff
@@ -49,23 +51,51 @@ func (db *BookDatabase) FetchBookList(ctx context.Context) ([]models.Book, error
  * For use after a user has selected a book to read
  */
 func (db *BookDatabase) FetchBookDetailsByID(ctx context.Context,
-	id string) (models.BookDetails, error) {
+	id string, lang string) (models.BookDetails, error) {
 
 	var book models.BookDetails
 
-	var query string = "SELECT id, title, author, image, read_video, read_body, " +
-		"explore_video, explore_body, learn_video, learn_body," +
+	var query string = "SELECT id, title, author, image, " +
 		"created_at FROM books WHERE id = $1"
 
+	var queryContent string = "SELECT read_video, read_body, explore_video, " +
+		"explore_body, learn_video, learn_body FROM book_contents WHERE id " +
+		"= $1 AND lang = $2"
+
 	err := db.Conn.QueryRowEx(ctx, query, nil, id).Scan(
-		&book.ID, &book.Title, &book.Author, &book.Image, &book.Read.Video,
-		&book.Read.Body, &book.Explore.Video, &book.Explore.Body,
-		&book.Learn.Video, &book.Learn.Body, &book.CreatedAt)
+		&book.ID, &book.Title, &book.Author, &book.Image,
+		&book.CreatedAt)
+
+	if err != nil {
+		return book, errors.Wrap(err, "error on SELECT FROM books in FetchBookByID")
+	}
+
+	err = db.Conn.QueryRowEx(ctx, queryContent, nil, id, lang).Scan(
+		&book.Read.Video, &book.Read.Body, &book.Explore.Video, &book.Explore.Body,
+		&book.Learn.Video, &book.Learn.Body)
 
 	if err != nil {
-		fmt.Printf("Error: %s\n", err)
 		return book, errors.Wrap(err, "error on SELECT FROM books in FetchBookByID")
 	}
 
 	return book, nil
 }
+
+/*
+ * Check for existence of a book with the specified id
+ */
+func (db *BookDatabase) CheckBookID(ctx context.Context, id string,
```

Comment:
Is there any reason you added this func instead of just making FetchBookDetailsById return a pointer?

I think this would be useful if there was any situation where we wanted to check for the existence of a book *without* fetching the book details right after, but I don't think we'll run into that situation. As of now the request needs to make 2 network calls which is just a little inefficient.